### PR TITLE
feat(CampaignsHistory): add pagination

### PR DESCRIPTION
**Pull request by fcoronelmakingsense**
- create new component to manage the pagination
- change service method to use the current page and the quantity per page
- change page when click in numbers

### CODE REVIEW

File: src/services/doppler-api-client.ts [hunk 1]
```diff
@@ -9,7 +9,8 @@ export interface DopplerApiClient {
   getSubscriber(email: string, apikey: string): Promise<ResultWithoutExpectedErrors<Subscriber>>;
   getSubscriberSentCampaigns(
     email: string,
-    apikey: string,
+    campaignsPerPage: number,
```
Comment (cbernat) [inline src/services/doppler-api-client.ts]: How about adding some test for this mappings?

File: src/services/doppler-api-client.double.ts [hunk 2]
```diff
@@ -49,34 +49,29 @@ const subscriber = {
...
-      campaignSubject: 'Ya comienza el dia 2. Accede a las conferencias',
-      deliveryStatus: 'softBounced',
-      clicksCount: 100,
-    },
-  ],
-  currentPage: 0,
-  itemsCount: 3,
-  pagesCount: 1,
-};
+const campaignDeliveryItems = [
```
Comment (cbernat) [inline src/services/doppler-api-client.double.ts]: I think here perhaps you could add more data, for us to see how this goes, visually. What do you think?

**Comment by cbernat** [comment]
So far so good. Very nice! 💪

File: src/services/doppler-api-client.ts (hunk 1 above)
Comment (fcoronelmakingsense) [reply src/services/doppler-api-client.ts]: i added

File: src/services/doppler-api-client.double.ts [hunk 3]
```diff
@@ -171,11 +166,19 @@ export class HardcodedDopplerApiClient implements DopplerApiClient {
...
   ): Promise<ResultWithoutExpectedErrors<CampaignDeliveryCollection>> {
-    console.log('getSubscriberSentCampaigns');
+    console.log('getSubscriberSentCampaigns', email, campaignsPerPage, currentPage);
     await timeout(1500);
 
+    const campaignDeliveryCollection = {
+      items: campaignDeliveryItems,
+      currentPage: currentPage,
+      itemsCount: campaignDeliveryItems.length,
+      pagesCount: 2,
```
[the remark this answers is not in the document]
Comment (fcoronelmakingsense) [reply src/services/doppler-api-client.double.ts]: added

File: src/services/doppler-api-client.double.ts (hunk 2 above)
Comment (fcoronelmakingsense) [reply src/services/doppler-api-client.double.ts]: yes i added now

File: src/services/doppler-api-client.double.ts [hunk 4]
```diff
@@ -49,35 +49,49 @@ const subscriber = {
...
+  return {
+    campaignId: index,
+    campaignName: 'Campaña estacional de primavera',
+    campaignSubject: '¿Como sacarle provecho a la primavera?',
+    deliveryStatus: getDeliveryStatus(Math.round(Math.random() * (5 - 1) + 1)),
+    clicksCount: Math.round(Math.random() * (100 - 1) + 1),
+  };
+});
+
+// const campaignDeliveryItems = [
```
Comment (cbernat) [inline src/services/doppler-api-client.double.ts]: Why do you have this commented code? are you going to use it for tests?

Comment (fcoronelmakingsense) [reply src/services/doppler-api-client.double.ts]: i saw it, its old! i already deleted

File: src/services/doppler-api-client.double.ts [hunk 5]
```diff
@@ -49,35 +49,29 @@ const subscriber = {
...
+    case 2:
+      return 'softBounced';
+    case 3:
+      return 'hardBounced';
+    default:
+      return 'notOpened';
+  }
 };
 
+const campaignDeliveryItems = [...Array(100)].map((_, index) => {
```
Comment (cbernat) [inline src/services/doppler-api-client.double.ts]: 😄

**Comment by cbernat** [comment]
I've tested this in cdn pointing to prod, and it's working great! 💪 https://cdn.fromdoppler.com/doppler-webapp/build2101/#/reports/campaigns-history?email=cbernat@makingsense.com&page=2

**Comment by cbernat** [comment]
If a user has no campaigns I see this
image

Could you please recheck?

**Pull request merged**